### ISSUE TITLE
No icons fallback

### ISSUE THREAD

**Issue by Vladimir-csp** [johanmalm/jgmenu#31]
I tried to run jgmenu when there was an icon theme set int tint2rc, but `launcher_icon_theme_override = 0` and no xsetting daemon running. Jgmenu refused to start with 
```
fatal: cache.c: icon_{theme,size} needs to be set
```
There should be some fallback, like using the theme from tint2 anyway if no xsettings is running, or look into GTK configs in XDG hierarchy.

**Comment by johanmalm** [comment]
I've stopped the unintentional die() for the time being. 
Agree with your suggestions.

**Comment by Vladimir-csp** [comment]
If you are going to implement fallbacks, a config option for shifting their priorities would be also nice.

**Comment by johanmalm** [comment]
I've now added 'src_icon_theme' and 'src_font' (see JGMENU-CONFIG(1) for explaination).
I've also removed 'ignore_xsettings' as it's now redundant.
I've only tested it on my BunsenLabs machine. Will log into some other distros tomorrow to test further.
Any feedback gratefully received as usual :-)

**Comment by Vladimir-csp** [comment]
That is a great feature. I think that 'j' is redundant and should just always be on top. If there explicitly is an icon or font theme in jgmenu config, it should override everything else.

**Comment by johanmalm** [comment]
I think you've made a good point. That would be cleaner.
I'll implement.

**Comment by johanmalm** [comment]
I have:
  - removed the 'j' in the two options
  - renamed them from src_* to *_fallback
  - made 'icon_theme' and 'font' take priority if set

Sorry for messing around over the last few commits. I should have thought it through before coding, but I think I'm happy with it now. The code is certainly cleaner and easier to follow as a result of these changes.

The old jgmenu-cache.sh did parse the gtk settings file as a fallback option. When we moved to the C-implementation, I dropped the support - and I have to admit I have missed it. I generally try not to worry too much about edge cases, but I think it's worth it for this one. I have three boxes on which I run Bunsenlabs, Arch and Ubuntu. The first two do not have xsettings running. With jgmenu reading the tint2 config file, I'm often running without a jgmenurc. Some tint2rcs (including the default created by tint2 on first start) do not have task_font and launcher_icon_theme defined. The gtk3.0 settings file therefore provides a useful fallback (particularly for the icon theme).
